README.md

```diff
@@ -64,6 +64,37 @@ List of repository sources:
 - [/flow-emulator](https://github.com/onflow/flow-emulator)
 - [/flow-cadut](https://github.com/onflow/flow-cadut)
 
-## Further help
+## Deployment
 
-Visit the [Nx Documentation](https://nx.dev) to learn more.
+⚠ Currently data cached in redis is not invalidated from new builds, so e.g. you might see cached mdx and markdown data from previous builds. to work around this, you need to manually remove keys from redis.
+
+Commits pushed to the `main` branch are automatically deployed to staging. Commits pushed to the `production` branch are automatically deployed to production. See [deploy.yml](.github/workflows/deploy.yml) for details.
+
+Recommended workflow to promote changes from `main` (staging) to production is:
+
+```bash
+git fetch
+git switch production
+git reset --hard origin/main
+git push --force
+```
+
+### Deployment takes forever or fails due to memory errors
+
+If deploys fail to go out, and the github actions hang indefinitely, a workaround to fix this is:
+
+1. Cancel the build
+1. Delete the fly app that is prefixed with `fly-builder-`, this can be done through CLI as shown below, or through the [fly dashboard](https://fly.io/dashboard/flow-docs)
+
+   ```bash
+   $ fly apps list
+   NAME                                    OWNER           STATUS  LATEST DEPLOY
+   flow-docs                               flow-docs       running 2022-06-22T00:07:40Z
+   flow-docs-staging                       flow-docs       running 7m45s ago
+   fly-builder-muddy-wildflower-4301       flow-docs       pending
+   $ fly apps destroy fly-builder-muddy-wildflower-4301
+   ...
+   ```
+
+1. Re-run the build by going to the latest commit on the appropriate branch, clicking "Re-run jobs", and then "Re-run failed jobs"
+2. Build should succeed
```

Comment:
@joekrill @alse @aureentuluvacal FYI forgot to go over this – basically sometimes you need to delete the "builder" app that fly spins up, deleting these builders is a safe operation